onnxruntime/core/framework/allocation_planner.cc

```diff
@@ -733,19 +733,22 @@ class PlannerImpl {
     // If the suggested_device is also CPU and default mem type, then
     // we check which one has higher alignment and use that one if it is so.
     // If the suggested device is CPU, but not the default mem type, then
-    // it is a CPU accessible memory device allocator. They typically have a page aligment
+    // it is a CPU accessible memory device allocator. They typically have a page alignment
     // so that would satisfy the alignment requirement of any other CPU consumers.
     // If one device is not on CPU, we default on the one that is CPU.
     auto determine_device = [](const OrtDevice& output_device, const OrtDevice& suggested_device) -> OrtDevice {
-      if (output_device.Type() == OrtDevice::CPU && suggested_device.Type() == OrtDevice::CPU) {
-        if (output_device.MemType() == OrtDevice::MemType::DEFAULT &&
-            suggested_device.MemType() == OrtDevice::MemType::DEFAULT) {
```

Comment:
The above comment mentions "default mem type". Is that still relevant or should it be updated?